src/components/DateRangePicker/SelectionPane.js

```diff
@@ -24,43 +26,53 @@ class SelectionPane extends React.Component {
     theme: themeShape
   };
 
+  componentDidMount() {
+    moment.locale(this.props.locale)
+  }
+
   _handleDateBoxClick = (date, selectedBox) => {
     this.props.onDateBoxClick(date, selectedBox);
   }
 
+  _formatDate = (date) => {
+    const d = moment.unix(date).format('MMM D, YYYY')
+
+    return d.charAt(0).toUpperCase() + d.slice(1)
+  }
+
   render () {
     const theme = StyleUtils.mergeTheme(this.props.theme);
     const styles = this.styles(theme);
-    const { selectedStartDate, selectedEndDate } = this.props;
+    const { getTranslation, selectedStartDate, selectedEndDate } = this.props;
 
     return (
       <div className='mx-selection-pane' style={styles.container}>
         <div>
-          <label style={styles.boxLabel}>From</label>
+          <label style={styles.boxLabel}>{getTranslation('From')}</label>
           <button
-            aria-label={`Select Start Date, ${selectedStartDate ? 'Current start date is ' + moment.unix(selectedStartDate).format('MMM D, YYYY') : ''}`}
+            aria-label={getTranslation(`Select Start Date${selectedStartDate ? ', Current start date is' : ''} %1`, this._formatDate(selectedStartDate))}
             className='mx-selection-pane-from-field'
             onClick={() => this._handleDateBoxClick(selectedStartDate, SelectedBox.FROM)}
             ref={this.props.getFromButtonRef}
             style={{ ...styles.dateSelectBox, ...this.props.selectedBox === SelectedBox.FROM ? styles.selectedDateSelectBox : {}}}
           >
-            {selectedStartDate ? moment.unix(selectedStartDate).format('MMM D, YYYY') : 'Select Start Date'}
+            {selectedStartDate ? this._formatDate(selectedStartDate) : getTranslation('Select Start Date')}
           </button>
 
-          <label style={styles.boxLabel}>To</label>
+          <label style={styles.boxLabel}>{getTranslation('To')}</label>
           <button
-            aria-label={`Select End Date, ${selectedEndDate ? 'Current end date is ' + moment.unix(selectedEndDate).format('MMM D, YYYY') : ''}`}
+            aria-label={getTranslation(`Select End Date${selectedEndDate ? ', Current end date is' : ''} %1`, this._formatDate(selectedEndDate))}
```

Comment:
Does this need a space before `selectedEndDate`? Not sure if the screenreader will read it out funny with it smashed together or not.

Reply:
I smooshed it because sometimes that string `Current end date` is there, and sometimes it isn't. I moved the space and the comma to inside that optional string, so that the screen reader won't show a hanging comma. 

previous without extra string:  `Select End Date, ` (note comma and extra space, even though this string is whole)
previous with extra string: `Select End Date, Current end ...`

updated without extra string: `Select End Date`
updated with extra string: `Select End Date, Current end ...`